### PR TITLE
Allow prometheus-formula only for SUSE systems

### PR DESCRIPTION
**Pull request by witekest**
Prometheus formula is currently supported only for SUSE systems.
The change hardens the formula to allow its application only to
supported systems. Additionally, the variables map can be extended by
the customer by setting the values in `prometheus:lookup` pillar.

Fixes: SUSE/spacewalk#17708

### CODE REVIEW

File: prometheus-formula/prometheus-formula.changes [hunk 1]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Thu May  5 08:03:42 UTC 2022 - Witek Bedyk <witold.bedyk@suse.com>
+
+- Version 0.6.2
+  * Allow prometheus-formula only for SUSE systems (bsc#1199149)
```
Comment (srbarrios) [inline prometheus-formula/prometheus-formula.changes:5]: This is not totally exact.

File: prometheus-formula/prometheus/init.sls [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {% from "prometheus/map.jinja" import prometheus with context %}
 
+{%- if prometheus %}
```
Comment (srbarrios) [inline prometheus-formula/prometheus/init.sls:3]: I guess that should cover all the supported systems :+1:

File: prometheus-formula/prometheus-formula.changes (hunk 1 above)
Comment (witekest) [reply prometheus-formula/prometheus-formula.changes:5]: That's a documentation bug. I'll provide a fix for that.
We support only SUSE for the monitoring server.

Comment (srbarrios) [reply prometheus-formula/prometheus-formula.changes:5]: Thanks to clarify, we will update the test suite accordly.

**Review by renner — APPROVED** [review]
It looks good to me, there is nothing to complain :+1: Thank you, Witek!

**Review by agraul — COMMENTED** [review]
I only have a question about the "pillar override": Could we add an example what a pilllar would need to look like? Without an example, users will need to read through `map.jinja` to know what to define.

**Comment by witekest** [comment]
> I only have a question about the "pillar override": Could we add an example what a pilllar would need to look like? Without an example, users will need to read through `map.jinja` to know what to define.

Good point. I'll add a README and describe that. For time reasons let me do that in the next PR though.

**Pull request merged**